src/Features/Lsif/Generator/Program.cs

```diff
@@ -53,21 +56,29 @@ private static async Task GenerateAsync(FileInfo? solution, FileInfo? project, F
             try
             {
                 // Exactly one of "solution", or "project" or "compilerInvocation" should be specified
-                if (solution != null && project == null && compilerInvocation == null)
+                var fileInputs = new[] { solution, project, compilerInvocation, binLog };
+                var nonNullFileInputs = fileInputs.Count(p => p is not null);
+
+                if (nonNullFileInputs != 1)
+                {
+                    throw new Exception("Exactly one of either a solution path, project path or a compiler invocation path should be supplied.");
+                }
+
+                if (solution != null)
                 {
                     await GenerateFromSolutionAsync(solution, lsifWriter, logFile);
                 }
-                else if (solution == null && project != null && compilerInvocation == null)
+                else if (project != null)
                 {
                     await GenerateFromProjectAsync(project, lsifWriter, logFile);
                 }
-                else if (solution == null && project == null && compilerInvocation != null)
+                else if (compilerInvocation != null)
                 {
                     await GenerateFromCompilerInvocationAsync(compilerInvocation, lsifWriter, logFile);
                 }
                 else
                 {
-                    throw new Exception("Exactly one of either a solution path, project path or a compiler invocation path should be supplied.");
+                    await GenerateFromBinaryLogAsync(binLog!, lsifWriter, logFile);
```

Comment:
should this be backed by an option to the tool?  seems odd that we'd just fallback to this.

Reply:
n/m.  misunderstanding.

Reply:
agree, or that the `binLog` is not null